### 1. 题目：有四个数字：1、3、5、7，能组成多少个互不相同且无重复数字的三位数？各是多少？

In [1]:
for i in range(1,8,2):
    for j in range(1,8,2):
        for k in range(1,8,2):
            if( i != k ) and (i != j) and (j != k):
                print(i*100+j*10+k)

135
137
153
157
173
175
315
317
351
357
371
375
513
517
531
537
571
573
713
715
731
735
751
753


### 2 . 题目：企业发放的奖金根据利润提成。利润计算方法如下：
* 利润在10万元以内的部分，奖金提成比例为10%；  
* 利润在10万至20万之间的部分，奖金提成比例为7.5%；  
* 利润高于20万元的部分，奖金提成比例为5%。  
#### 利用input函数从键盘输入当月利润I（万元），求应发放奖金总数？ 

In [2]:
i = float(input('请输入净利润(万元):'))
arr = [20,10,0]
rat = [0.05,0.075,0.1]
r = 0
for idx in range(0,3):
    if i>arr[idx]:
        r+=(i-arr[idx])*rat[idx]
        print((i-arr[idx])*rat[idx])
        i=arr[idx]
print("总奖金:", r)

请输入净利润(万元):100
4.0
0.75
1.0
总奖金: 5.75


### 3. 题目：输入某个年份，判断这一年有多少天？

In [3]:
year = int(input('year:\n'))

#判断闰年：（1）能被4整除但不能被100整除（2）能被400整除
if(year % 4 == 0 and year % 100 != 0) or (year % 400 == 0):
    print("这一年有366天")
else:
    print("这一年有365天")

year:
2024
这一年有366天


### 4. 题目：输入三个整数x,y,z，请把这三个数由小到大输出。

In [4]:
l = []
for i in range(3):
    x = int(input('integer:\n'))
    l.append(x)
l.sort()
print(l)

integer:
4
integer:
4
integer:
9
[4, 4, 9]


### 5. 题目：构造斐波那契数列，并输出数列中的第10个数。

In [3]:
def fib(n):
    a,b = 1,1
    for i in range(n-1):
        a,b = b,a+b
    return a
 
# 输出了第10个斐波那契数列
print(fib(10))

55


In [4]:
# 使用递归
def fib2(n):
    if n==1 or n==2:
        return 1
    return fib2(n-1)+fib2(n-2)
 
# 输出了第10个斐波那契数列
print(fib2(10))

55


### 6. 题目：输出 9*9 乘法口诀表。

In [1]:
for i in range(1, 10):
    
    for j in range(1, i+1):
        print("%d * %d = %2d" % (i, j, i*j), end="; ")
    print("\b\b")

1 * 1 =  1
2 * 1 =  2; 2 * 2 =  4
3 * 1 =  3; 3 * 2 =  6; 3 * 3 =  9
4 * 1 =  4; 4 * 2 =  8; 4 * 3 = 12; 4 * 4 = 16
5 * 1 =  5; 5 * 2 = 10; 5 * 3 = 15; 5 * 4 = 20; 5 * 5 = 25
6 * 1 =  6; 6 * 2 = 12; 6 * 3 = 18; 6 * 4 = 24; 6 * 5 = 30; 6 * 6 = 36
7 * 1 =  7; 7 * 2 = 14; 7 * 3 = 21; 7 * 4 = 28; 7 * 5 = 35; 7 * 6 = 42; 7 * 7 = 49
8 * 1 =  8; 8 * 2 = 16; 8 * 3 = 24; 8 * 4 = 32; 8 * 5 = 40; 8 * 6 = 48; 8 * 7 = 56; 8 * 8 = 64
9 * 1 =  9; 9 * 2 = 18; 9 * 3 = 27; 9 * 4 = 36; 9 * 5 = 45; 9 * 6 = 54; 9 * 7 = 63; 9 * 8 = 72; 9 * 9 = 81


### 7. 题目：输入2024年的任何一个格式为“YYYY-MM-DD HH:MM:SS”的时间，计算其是2024年的第几秒。（以“2024-01-01 00:00:01”为第1秒）

In [2]:
from datetime import datetime

input_time = input("请输入2024年的时间[YYYY-MM-DD HH:MM:SS]：")
year, month, day = map(int, input_time.split()[0].split("-"))
hour, minute, second = map(int, input_time.split()[1].split(":"))

start_time = datetime(year,1,1,0,0,0) #2024-01-01 00:00:00
end_time = datetime(year,month,day,hour,minute,second) 
time_gap = end_time - start_time

print("输入时间为2024年的第%d秒" % int(time_gap.total_seconds()))

输入时间为2024年的第2061秒


### 8. 题目：判断101-200之间有多少个素数，并输出所有素数。

In [9]:
count = 0
leap = 1
from math import sqrt
for m in range(101,201):
    k = int(sqrt(m + 1))
    for i in range(2,k + 1):
        if m % i == 0:
            leap = 0
            break
    if leap == 1:
        print('%-4d' % m)
        count += 1
        if count % 10 == 0:
            print()
    leap = 1
print('The total is %d' % count)

101 
103 
107 
109 
113 
127 
131 
137 
139 
149 

151 
157 
163 
167 
173 
179 
181 
191 
193 
197 

199 
The total is 21


### 9. 题目：打印出所有的"水仙花数"，所谓"水仙花数"是指一个三位数，其各位数字立方和等于该数本身。例如：153是一个"水仙花数"，因为$153=1^3＋5^3＋3^3$。

In [10]:
for n in range(100,1000):
    i = int(n / 100)
    j = int(n / 10 % 10)
    k = int(n % 10)
    if n == i ** 3 + j ** 3 + k ** 3:
        print(n)

153
370
371
407


In [11]:
print(i,j,k)

9 9 9


### 10. 题目：将一个正整数分解质因数。例如：输入90，则打印出90=2 * 3 * 3 * 5。

程序分析：对n进行分解质因数，应先找到一个最小的质数k，然后按下述步骤完成：
(1)如果这个质数恰等于n，则说明分解质因数的过程已经结束，打印出即可。
(2)如果n<>k，但n能被k整除，则应打印出k的值，并用n除以k的商,作为新的正整数你n,重复执行第一步。
(3)如果n不能被k整除，则用k+1作为k的值,重复执行第一步。

程序源代码：

In [12]:
def reduceNum(n):
    print('{} = '.format(n),end='')
    if not isinstance(n, int) or n <= 0 :
        print('请输入一个正确的数字 !')
        exit(0)
    elif n in [1] :
        print('{}'.format(n))
    while n not in [1] : # 循环保证递归
        for index in range(2, int(n) + 1) :
            if n % index == 0:
                n /= index # n 等于 n/index
                if n == 1: 
                    print(index,end='')
                else : # index 一定是素数
                    print('{} * '.format(index),end='')
                break
reduceNum(1)
reduceNum(100)

1 = 1
100 = 2 * 2 * 5 * 5